packages/components/src/components/gif.tsx

```diff
@@ -149,11 +149,13 @@ const Gif = ({
 
     const trackWithMoat = async () => {
         if (showGif && container.current) {
-            await moatLoader
             const { bottle_data: bottleData } = gif
             const moatCompatibleData = constructMoatData(bottleData as any)
             if (moatCompatibleData) {
-                moatAdNumber.current = moat.startTracking(container.current, moatCompatibleData)
+                await moatLoader
```

Comment:
Only load moat if we have `moatCompatibleData`, and then after the `await`, also check the container again 

Will save us the request when we're not using moat, and could potentially avoid passing null to moat.startTracking. That would have only happen though if while the `moatLoader` was being fetched, the gif component was unmounted (pretty unlikely)